pkg/operator/statefulset.go

```diff
@@ -75,13 +75,38 @@ func CreateConfigReloader(
 		Limits:   v1.ResourceList{},
 		Requests: v1.ResourceList{},
 	}
-	if config.CPU != "0" {
-		resources.Limits[v1.ResourceCPU] = resource.MustParse(config.CPU)
-		resources.Requests[v1.ResourceCPU] = resource.MustParse(config.CPU)
+
+	// Prioritize specific flags over generic ones
```

Comment:
Lets add some tests for these use cases, unit tests is fine maybe we can separate it out to its standalone function, thanks!